### PR TITLE
Add support for Strings claim

### PR DESCRIPTION
**Pull request by stefan-improbable**
This fixes https://github.com/gambol99/keycloak-proxy/issues/314

### CODE REVIEW

File: middleware.go [hunk 1]
```diff
@@ -211,6 +212,60 @@ func (r *oauthProxy) authenticationMiddleware(resource *Resource) func(http.Hand
...
 
+// checkClaim checks whether claim in userContext matches claimName, match. It can be String or Strings claim.
+func (r *oauthProxy) checkClaim(user *userContext, claimName string, match *regexp.Regexp, resourceURL string) bool {
+	errFields := []zapcore.Field{
+		zap.String("claim", claimName),
+		zap.String("access", "denied"),
+		zap.String("email", user.email),
+		zap.String("resource", resourceURL),
+	}
+
```
Comment (gambol99) [inline middleware.go]: given [Claims](https://github.com/gambol99/go-oidc/blob/master/jose/claims.go#L10) is `type Claims map[string]interface{}` .. we could just and fail quick .. What do you think?

```
if _, found := user.claims[claimName]; !found {
    r.log.Warn("the token does not have the claim", errFields...)
    return false
}

Comment (stefan-improbable) [reply middleware.go]: Yep, I've added the statement here.

**Comment by stefan-improbable** [comment]
@gambol99 I've added some info to README

**Comment by gambol99** [comment]
cheers @stefan-improbable  ... LGTM

**Comment by stefan-improbable** [comment]
@gambol99 thanks for the quick review!

**Pull request merged**